src/core/thread/router_table.hpp

```diff
@@ -338,6 +338,27 @@ class RouterTable : public InstanceLocator, private NonCopyable
      */
     IteratorBuilder Iterate(void) { return IteratorBuilder(GetInstance()); }
 
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    void GetAllowedRouterIdRange(uint8_t *aMinRouterIdAllowed, uint8_t *aMaxRouterIdAllowed) const
```

Comment:
Use reference variable instead of pointer here?
I think we recommend reference use over pointer in c++ core API in our style guide.